### PR TITLE
[INLONG-4653][CI] Add concurrency support on GitHub Actions

### LINKED ISSUE #4653: [Improve][CI] Add concurrency support on GitHub Actions
### Description

Add concurrency support on GitHub Actions.
See https://docs.github.com/en/actions/using-jobs/using-concurrency for more information.

### InLong Component

Other for not specified component

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by shink**
Fixes #4653

### Motivation

Add concurrency support on GitHub Actions

### Modifications

Add concurrency support on GitHub Actions.
See https://docs.github.com/en/actions/using-jobs/using-concurrency for more information.

### CODE REVIEW

**Comment by dockerzhang** [comment]
@shink please rebase from master firstly, thanks.
image

**Comment by shink** [comment]
@dockerzhang OK.
BTW, I don't think the `master` branch can be force updated.

File: .github/workflows/ci_docker.yml [hunk 1]
```diff
@@ -90,7 +97,10 @@ jobs:
           fi
 
       - name: Push Docker images to Docker Hub
-        if: ${{ success() && steps.match-branch.outputs.match == 'true' }}
+        if: |
+          success()
+          && steps.check-workflow-diff.outputs.changed_only == 'no'
+          && steps.match-branch.outputs.match == 'true'
```
Comment (shink) [inline .github/workflows/ci_docker.yml:103]: If only the workflow file (`.github/workflows/ci_docker.yml`) is changed, there is no need to publish images.

**Comment by shink** [comment]
@dockerzhang I know. I mean the `master` branch should not be force-updated. It will cause additional work for all developers. This is a default norm.

File: .github/workflows/ci_docker.yml [hunk 2]
```diff
@@ -73,6 +77,13 @@ jobs:
         env:
           CI: false
 
+      # Check if only the workflow file is changed.
+      - name: Check workflow diff
+        id: check-workflow-diff
+        uses: apache/pulsar-test-infra/diff-only@master
+        with:
+          args: .github/workflows/ci_docker.yml
+
```
Comment (shink) [inline .github/workflows/ci_docker.yml:86]: The `apache/pulsar-test-infradiff-only` action is used here. See https://github.com/apache/pulsar-test-infra/tree/master/diff-only for more information.

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**